xsamtools/misc_utils.py

```diff
@@ -0,0 +1,17 @@
+import subprocess
+
+from typing import Any
+
+
+def subprocess_w_stderr_run(cmd: Any, stdout: Any = subprocess.PIPE, check: bool = False, shell: bool = True):
```

Comment:
- This can be renamed to just "run"
- The slew of parameters can be avoided while exposing the entire parameter space of `subprocess.run`:
```
def run(cmd: Any, check: bool=True, capture_output: bool=True, **kwargs) -> subprocess.CompletedProcess:
    kwargs['check'] = check
    kwargs['capture_output'] = capture_output
    ...
```
- please avoid using `shell=True`. It is not used anywhere else in xsamtools.
- note that `capture_output=True` is more expressive than the arcane `stderr=subprocess.PIPE`